### PR TITLE
Add logs and metrics to default features in api, sdk and otlp crate

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes https://github.com/open-telemetry/opentelemetry-rust/issues/1709
All 3 signals are planned to reach beta in the next release. This PR is changing the default for api,sdk,otlp crates to include all three signals by default. The separate feature flags are still maintained, so if anyone wants to only pick only, it is still possible.

### CODE REVIEW

File: examples/logs-basic/README.md [hunk 1]
```diff
@@ -1,16 +1,17 @@
-# Log Appender for API -  Example
+# OpenTelemetry Log Appender for log -  Example
```
Comment (cijothomas) [inline examples/logs-basic/README.md:1]: sorry, unrelated changes was sitting in my local! Hope this is okay.

**Review by lalitb — APPROVED** [review]
Thanks. Good to check if we need to add this for opentelemetry-proto too.

**Comment by cijothomas** [comment]
> Thanks. Good to check if we need to add this for opentelemetry-proto too.

I left that as it is not clear why would end users need that directly...
